tests/test_local_move.py

```diff
@@ -64,6 +64,14 @@ def assert_no_drift(
 
     expected_selection_fraction_traj = np.array([observable_fxn(x) for x in traj])
 
+    # A sanity check that the early samples don't have a massive jump within
+    # in the case of unstable local MD this test can pass neighboring atoms go from ~10% of the system to ~100% of the system
+    # in the first step
+    differences_early = np.diff(expected_selection_fraction_traj[:n_samples]) / expected_selection_fraction_traj[0]
+    assert (
+        np.all(differences_early) < 2.0
+    ), "Difference between first and last sample greater than 200%, likely unstable"
+
```

Comment:
I don't quite understand the motivation for this assertion -- (1) did this come up when testing including this term, (2) can this assertion be combined with the next assertion?

(this one: check that (f[t] - f[t-1] / f[0]) < 2 for all t. next assertion: compare first and second half of trajectory, check that these don't differ by more than 50% up or down. could maybe modify the second assertion to average over a smaller initial portion of the trajectory.)

Reply:
This came up in testing, but I had seen similar behavior much earlier.

The issue is that occasionally it will be stable for a 'few' steps, but then suddenly explode, resulting in the number of selected particles going from say 1000 to 6000.

Since assertion is looking at the beginning as it verifies that you aren't introducing big differences by switching to local moves. If you are, it seems like it warrants more investigation.

Perhaps doing a running average on the start to make sure the values agree would be better.

Reply:
Ahh, thanks for the additional context!

This does indicate a limitation of the existing assertion (averaging over too large of an initial window) -- perhaps in the future this could be made more sensitive (similar to the new assertion in this PR), by averaging over a smaller initial window.

For now there's value in having both.

Reply:
Minor improvement to the comment and adds abs: https://github.com/proteneer/timemachine/pull/1014/commits/aa47b6aa5f9f4da2769e280d80400c9c730017b2